src/ZkFold/Symbolic/Compiler/ArithmeticCircuit/Instance.hs

```diff
@@ -133,9 +135,16 @@ instance {-# OVERLAPPING #-} SymbolicData a x => Conditional (Bool (ArithmeticCi
 
 -- TODO: make a proper implementation of Arbitrary
 instance Arithmetic a => Arbitrary (ArithmeticCircuit a) where
-    arbitrary = do
-        let ac = mempty { acOutput = 1} * mempty { acOutput = 2}
-        return ac
+    arbitrary = frequency [
+        (1, return mempty { acOutput = 1}),
+        (2, liftM2 (+) arbitrary arbitrary),
+        (2, liftM2 (-) arbitrary arbitrary),
+        (2, liftM2 (*) arbitrary arbitrary),
+        (2, liftM2 (//) arbitrary arbitrary),
+        (2, fromConstant <$> c),
+        (2, boolCheckC <$> arbitrary) ]
```

Comment:
This adds a bool check on a variable. The problem is that if this variable is intermediate, i.e. it is computed using input variables and constants, the circuit is very likely cannot be satisfied. I suggest applying it only to input variables (should be similar to the first case).